cmake/host-tools.cmake

```diff
@@ -29,6 +29,24 @@ if(${CMAKE_MATCH_1} VERSION_LESS ${MIN_DTC_VERSION})
   )
 endif()
 
+# Usage:
+#   check_dtc_flag("-Wtest" DTC_WARN_TEST)
+#
+# will set DTC_WARN_TEST if "-Wtest" is a valid flag for the version of dtc we
+# using
+function(check_dtc_flag flag var)
```

Comment:
We are rightly re-using the CMake naming scheme:

check_${cli_tool}_flag(flag var).

But we have changed the semantics of 'var'. We should align, and have var contain a boolean value instead of the value of flag.

Reply:
Do you have an example?

Reply:
https://cmake.org/cmake/help/latest/module/CheckCXXCompilerFlag.html

https://github.com/zephyrproject-rtos/zephyr/blob/master/cmake/extensions.cmake#L991

Reply:
Made the changes to use a check var